src/utils/pgwire/src/pg_extended.rs

```diff
@@ -155,13 +172,29 @@ impl PgStatement {
         // "SELECT 'A', 'B', 'C'".
         let instance_query_string = replace_params(statement, &generic_params, &params);
 
-        // 4. Create a new portal.
-        PgPortal {
+        // 4. Get row_description and return portal.
+        let row_description = if instance_query_string.starts_with("select")
+            || instance_query_string.starts_with("SELECT")
```

Comment:
This part of code is very similar to above.

Reply:
We can extract a function inside portal?
`fn infer_row_description(sql: &str,session: Arc<SM::Session>,) -> Result<Vec<PgFieldDescriptor>, ()>`